### DISCUSSION TITLE
Package specific update rules - delayed package updates

### DISCUSSION THREAD

**Discussion by sebarthel89** [renovatebot/renovate#22175] [Suggest an Idea]
**What would you like Renovate to be able to do?**
I would like to have 2 packages for the same dependency using 2 different versions.

My package json dependencies will look like:

```
"react": "16.11.0",
"react-experimental": "npm:react@16.12.0",
```

The target is to try manage the newest version of a package using "package-experimental". In my application itself I will make sure that only a few people will use the experimental version. 

Is it somehow possible to tell renovate to not upgrade to the newest version ? For example

```
packageRules: {
   packageNames: ['react'],
   minorVersionUpgradeDelay: 1
},
packageRules: {
   packageNames: ['react-experimental'],
   minorMajorVersionUpgradeDelay: 0
}
```

This would make surer that:
- the package reference syntax will be kept and properly changed on version upgrades ("react-experimental": "npm:react@16.12.0",
)
- minorMajorVersionUpgradeDelay (naming is hard) will make sure that the non experimental version is never updated to the newest version. If there is a new version 16.14.0 available then the version will be changed to 16.13.0. In case that there is a new version 17.0.0 available then iti will be upgraded to 16.12.x coming from 16.11.0 . And as soon as 17.1.0 is available the version will be upgraded to 17.0.x.
- hotfix versions should in the case always be applied.

**Comment by rarkins** [comment]
First of all, you can already have two packages with same dependency on different versions, and define different rules for Renovate for each. Renovate should support upgrading these and maintaining the syntax. Add a separate issue/bug report if it doesn't work for you. Essentially your `react-experimental` approach should behave that way as-is without any configuration required.

But we don't support any concept yet of a "minor delay". I guess another way you could describe it is a "release lag", right? You want to lag one minor release.

**Comment by sebarthel89** [comment]
+1 for the naming `release lag` :)

**Comment by HonkingGoose** [comment]
This issue seems related:

- https://github.com/renovatebot/renovate/issues/12914
